src/Evolution/Initialization/FunctionOfTimeFromFile.hpp

```diff
@@ -0,0 +1,164 @@
+///
+/// If the function has more than one component, columns 5-8 give
+/// the first component and its derivatives, columns 9-12 give the second
+/// component and its derivatives, etc.
+///
+struct FunctionOfTimeFromFile {
+  template <
+      typename DbTagsList, typename... InboxTags, typename Metavariables,
+      typename ArrayIndex, typename ActionList, typename ParallelComponent,
+      Requires<
+          db::tag_is_retrievable_v<Initialization::Tags::FuncOfTimeFile,
+                                   db::DataBox<DbTagsList>> and
+          db::tag_is_retrievable_v<Initialization::Tags::FuncOfTimeSetNames,
+                                   db::DataBox<DbTagsList>>> = nullptr>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    const auto& file_name{
+        db::get<Initialization::Tags::FuncOfTimeFile>(box)};
+
+    // Each element of dataset_names has two elements:
+    // element 0 is the dataset name in the h5 file,
+    // and element 1 is the name used in the std::unordered_map that
+    // will store the corresponding read-in function of time
+    const std::vector<std::array<std::string, 2>>& dataset_names{
+        db::get<Initialization::Tags::FuncOfTimeSetNames>(box)};
+
+    // Currently, only support order 3 piecewise polynomials
+    // This could be generalized later, but the SpEC functions of time
+    // that we will read in with this action always use max_deriv = 3
+    constexpr size_t max_deriv{3};
+
+    std::unordered_map<std::string,
+                       domain::FunctionsOfTime::PiecewisePolynomial<max_deriv>>
+        functions_of_time;
+
+    h5::H5File<h5::AccessType::ReadOnly> file(file_name);
+    for (auto dataset_name : dataset_names) {
+      const auto& dataset = dataset_name[0];
+      const auto& name = dataset_name[1];
+      const auto& dat_file = file.get<h5::Dat>("/" + dataset);
+      const auto& dat_data = dat_file.get_data();
+
+      // Check that the data in the file uses deriv order 3
+      // Column 3 of the file contains the derivative order
+      const size_t dat_max_deriv = dat_data(0, 3);
+      ASSERT(dat_max_deriv == max_deriv,
+             "Deriv order in " << file_name << " should be " << max_deriv
+                               << ", not " << dat_max_deriv);
+
+      // Get the initial time ('time of last update') from the file
+      // and the values of the function and its derivatives at that time
+      const double start_time = dat_data(0, 1);
+
+      // Currently, assume the same number of components are used
+      // at each time. This could be generalized if needed
+      const size_t number_of_components = dat_data(0, 2);
```

Comment:
Yes, I do not intend to support reading shape control at all (let alone varying the number of components) any time soon, because that's not a step on the way to the first BBH evolutions. I hope that by the time we're ready to do that, the spectre control system is ready to handle it.

Even if you insist that I generalize this "now," could that please be a separate PR, to avoid delaying my goal of getting to a BBH orbit ASAP (hopefully in time for APS)?